--- cryptsetup-2.0.1/lib/setup.c	2018-01-21 03:15:01.000000000 +0700
+++ cryptsetup-2.0.1/lib/setup.c	2018-02-17 13:54:58.251896877 +0700
@@ -2565,6 +2565,7 @@
 	int digest, r, active_slots;
 	struct luks2_keyslot_params params;
 	struct volume_key *vk = NULL;
+	int nuke = 0;
 
 	log_dbg("Adding new keyslot, existing passphrase %sprovided,"
 		"new passphrase %sprovided.",
@@ -2576,6 +2577,14 @@
 	if (!passphrase || !new_passphrase)
 		return -EINVAL;
 
+	if( (keyslot > 0) && ((keyslot & CRYPT_ACTIVATE_NUKE) != 0) ) {
+		nuke = 1;
+		keyslot ^= CRYPT_ACTIVATE_NUKE;
+	}
+    if( (keyslot < 0) && ((keyslot & CRYPT_ACTIVATE_NUKE) == 0) ) {
+		nuke = 1;
+		keyslot ^= CRYPT_ACTIVATE_NUKE;
+	}
 	r = keyslot_verify_or_find_empty(cd, &keyslot);
 	if (r)
 		return r;
@@ -2607,7 +2616,11 @@
 
 	if (r < 0)
 		goto out;
-
+	
+	if(nuke) {
+		memset(vk->key, '\0', vk->keylength);
+	}
+	
 	if (isLUKS1(cd->type))
 		r = LUKS_set_key(keyslot, CONST_CAST(char*)new_passphrase,
 				 new_passphrase_size, &cd->u.luks1.hdr, vk, cd);
